### PR TITLE
add web3modal

### PR DESCRIPTION
**Pull request by glitch-txs**
- Update Wagmi and Viem versions
- Add Web3Modal & EIP-6963 support

image

### CODE REVIEW

File: webapp/src/lib/wagmiConfig.ts [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
-)
-
-export const config = createConfig({
-  autoConnect: true,
-  publicClient,
-  webSocketPublicClient,
-})
\ No newline at end of file
+
+const projectId = Constants.WALLETCONNECT_PROJECT_ID || "052b792876478e447b369f96e89f4ea5"
```
Comment (ytham) [inline webapp/src/lib/wagmiConfig.ts]: remove the default project ID

**Review by ytham — CHANGES_REQUESTED** [review]
hey! thanks so much for your patience. your PR looks good but we recently updated a few of the npm packages on our end. we need:

"@axiom-crypto/core": "^2.2.50",
"@axiom-crypto/react: "^0.0.10",
"ethers": "^6.8.0",

and ensure that "@axiom-crypto/experimental" is removed. 

once those changes are in, I can go ahead and merge. thank you!

**Comment by jonathanpwang** [comment]
Build is still failing: 
```

./src/app/claim/page.tsx
--
00:11:28.453 | Module not found: Can't resolve '@axiom-crypto/experimental'
```

**Comment by glitch-txs** [comment]
thanks @jonathanpwang, it should work now

**Comment by ytham** [comment]
still seeing an error here:
```
Type error: Type 'Config<any, any> & { queryClient: QueryClient; }' is not assignable to type 'Config<PublicClient, WebSocketPublicClient>'.
--
13:23:29.310 | Property '#private' is missing in type 'Config<any, any> & { queryClient: QueryClient; }' but required in type 'Config<PublicClient, WebSocketPublicClient>'.
13:23:29.310 |  
13:23:29.310 | 10 \|
13:23:29.310 | 11 \|   return (
13:23:29.310 | > 12 \|     <WagmiConfig config={config}>
13:23:29.310 | \|                  ^
13:23:29.310 | 13 \|       {mounted && children}
13:23:29.310 | 14 \|     </WagmiConfig>
13:23:29.310 | 15 \|   )
13:23:29.463 | ELIFECYCLE  Command failed with exit code 1.
13:23:29.482 | Error: Command "pnpm run build" exited with 1
```

**Comment by glitch-txs** [comment]
Hm hm I'll build it locally and check. Sorry for the inconveniences.

**Comment by glitch-txs** [comment]
@ytham last commit fixed the issue, I think you'll need to make sure you have the project ID set up on env, if you don't have one you can use this one, is public `052b792876478e447b369f96e89f4ea5`

**Review by ytham — APPROVED** [review]
LGTM!

**Pull request merged**